### PR TITLE
refactor(modal): updated headers and spacing

### PR DESCRIPTION
**Pull request by julio8a**
**Related Issue:** #1500

## Summary

- Updated header text for s, m, l
- Updated all spacing to match Figma

Working from the [Figma/Tailwind components refactor project](https://github.com/Esri/calcite-components/projects/14)

https://github.com/Esri/calcite-components/projects/14#card-60668030

### CODE REVIEW

**Review by asangma — CHANGES_REQUESTED** [review]
Just one request.

File: src/components/calcite-modal/calcite-modal.scss [hunk 1]
```diff
@@ -110,16 +119,17 @@
...
-  padding: var(--calcite-modal-padding-large);
+  padding: var(--calcite-modal-padding) var(--calcite-modal-padding-large);
 }
 
 .content--no-footer {
   @apply rounded-b;
 }
 
 @include slotted("content", "*") {
+  @apply leading-6;
```
[the remark this answers is not in the document]
Comment (asangma) [reply src/components/calcite-modal/calcite-modal.scss:130]: cc @mitc7862

**Comment by bstifle** [comment]
@julio8a can we drop the X on Small to 16x16 px icon?

File: src/components/calcite-modal/calcite-modal.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (julio8a) [reply src/components/calcite-modal/calcite-modal.scss:130]: Yeah, short-term, I can remove this line height.

**Comment by julio8a** [comment]
Just pushed some updates:
- small modal 'x' uses scale=s of calcite-icon
- removed line-height from slotted content

**Comment by julio8a** [comment]
Okay, I think this is ready for another review.

**Review by bstifle — APPROVED** [review]
PERFECT

**Review by asangma — APPROVED** [review]
Thanks @julio8a! 

🧹 ✨

**Pull request merged**